lib/Controller/TemplateFieldController.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Richdocuments\Controller;
+
+use OCA\Richdocuments\Service\TemplateFieldService;
+use OCA\Richdocuments\TemplateManager;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Attribute\NoAdminRequired;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\IRequest;
+
+class TemplateFieldController extends OCSController {
+	private TemplateFieldService $templateFieldService;
+	private TemplateManager $templateManager;
+
+	/**
+	 * Template fields controller
+	 *
+	 * @param string $appName,
+	 * @param IRequest $request,
+	 * @param TemplateFieldService $templateFieldService
+	 * @param TemplateManager $templateManager
+	 */
+	public function __construct(
+		string $appName,
+		IRequest $request,
+		TemplateFieldService $templateFieldService,
+		TemplateManager $templateManager
+	) {
+		parent::__construct($appName, $request);
+
+		$this->templateFieldService = $templateFieldService;
+		$this->templateManager = $templateManager;
+	}
+
+	/**
+	 * @param int $fileId
+	 * @return DataResponse
+	 */
+	#[NoAdminRequired]
+	public function extractFields(int $fileId): DataResponse {
+		try {
+			$fields = $this->templateFieldService->extractFields($fileId);
+
+			return new DataResponse($fields, Http::STATUS_OK);
+		} catch (\Exception $e) {
+			return new DataResponse([$e->getMessage()], Http::STATUS_INTERNAL_SERVER_ERROR);
```

Comment:
Sorry should have been more clear, ideally just return a more generic error message instead of sending the exception message to the response as it might contain and leak internal server data.